nixos/modules/services/networking/firewall.nix

```diff
@@ -177,6 +177,21 @@ let
       ) cfg.allowedUDPPortRanges
     }
 
+    # Accept packets on the extra allowed host+port combination.
+    ${concatMapStrings (allowed:
+      if (allowed.useIPv4 || allowed.usedIPv6) then (
+        (if allowed.useIPv4 then (
+           if allowed.useIPv6 then ("ip46tables -A nixos-fw ") else ("iptables -A nixos-fw ")
+        ) else (
+           "ip6tables -A nixos-fw "
+        ))
+        + (if (allowed.source != "") then (" --source " + allowed.source) else "")
```

Comment:
Please use `optionalString` and string interpolation in all instances instead of `if` clauses and `+`

ex.:
```
${optionalString (allowed.source != "") " --source ${allowed.source}"}
```